### PR TITLE
fix poly drawing with smarter methods for single colors and rects

### PR DESCRIPTION
**Pull request by jkrumbiegel**
this brings down svg saving time for a couple of kde polys with 100 or so vertices to 80ms from a couple seconds for me. meshes seem to be quite slow, also they are overkill for polys with their single color

### CODE REVIEW

File: src/CairoMakie.jl [hunk 1]
```diff
@@ -494,6 +494,35 @@ function draw_plot(scene::Scene, screen::CairoScreen, primitive::Combined)
...
+    rgba = RGBA(c)
+    red(rgba), green(rgba), blue(rgba), alpha(rgba)
+end
+
+"""
+Special method for polys so we don't fall back to atomic meshes, which are much more
+complex and slower to draw than standard paths with single color.
+"""
+function draw_plot(scene::Scene, screen::CairoScreen, poly::Poly)
+    if poly.visible[] == false
```
Comment (asinghvi17) [inline src/CairoMakie.jl:507]: Is this needed as a special case, or would it be better to check for `visible` even in non-atomic plot types?

File: src/CairoMakie.jl [hunk 2]
```diff
@@ -494,6 +494,35 @@ function draw_plot(scene::Scene, screen::CairoScreen, primitive::Combined)
...
+
+    model = Mat4f0(I)
+    atomic_lines = poly.plots[2]
+    points = project_position.(Ref(scene), atomic_lines[1][], Ref(model))
+    Cairo.move_to(screen.context, points[1]...)
+    for p in points[2:end]
+        Cairo.line_to(screen.context, p...)
+    end
+    Cairo.close_path(screen.context)
+    Cairo.set_source_rgba(screen.context, rgbatuple(to_color(poly.color[]))...)
```
Comment (asinghvi17) [inline src/CairoMakie.jl:519]: GLMakie polys support continuous color - see http://juliaplots.org/MakieReferenceImages/gallery//poly_and_colormap/index.html.  We could support this by using Cairo's mesh gradients, but at that point you probably lose quite a bit of performance.  For what it's worth, colors are per vertex, so the mesh method might not be too bad...

Comment (asinghvi17) [reply src/CairoMakie.jl:519]: We could also fall back to `mesh` if `color isa AbstractArray`...

**Comment by jkrumbiegel** [comment]
> Looks good to me! One feature which we'll eventually need is hole support, but that can come later.

Polygons don't have holes though (maybe degenerate polys with self intersections, but not in the way you mean). So I think we should just add a Path primitive to AbstractPlotting and CairoMakie. This would include bezier segments, and ClosePath statements. With those you can then make holes and whatever you please.

**Comment by asinghvi17** [comment]
I just realized - the mesh fallback will already account for this, so it should be OK.

**Comment by jkrumbiegel** [comment]
tests pass now except the `band` one, which is only because of a different color preset as far as I can see. Nothing about the poly code

**Pull request merged**